From 8a4a9b82d4f8fd39db8417fd06d969505fde9d0a Mon Sep 17 00:00:00 2001
From: Rodrigo Rebello <rprebello@gmail.com>
Date: Tue, 22 Dec 2015 18:02:22 -0200
Subject: [PATCH 1/1] configure.ac: add option to disable image output

Even when the gd library is present, it may be desirable to not build
'vnstati' as image output is an optional feature.

Also, when testing for the gd library, use gdImagePng() instead of
gdImageLine() since it's possible that the installed gd library doesn't
have PNG support. In such cases, the test in the configure script passed
(because gdImageLine() is always present), but the build failed with:

  src/vnstati.o: In function `writeoutput':
  vnstati.c:(.text+0x3fc): undefined reference to `gdImagePng'
  collect2: error: ld returned 1 exit status

Upstream status: accepted, not yet released.
https://github.com/vergoh/vnstat/pull/35

Signed-off-by: Rodrigo Rebello <rprebello@gmail.com>
---
 configure.ac | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index be3d449..89f71ef 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,10 +13,16 @@ AM_PROG_CC_C_O
 AC_PROG_MAKE_SET
 AC_PROG_MKDIR_P
 
+AC_ARG_ENABLE([image-output], [
+AS_HELP_STRING([--disable-image-output], [disable PNG image output])])
+
 # Checks for libraries.
 AC_CHECK_LIB([m], [pow])
-AC_CHECK_LIB([gd], [gdImageLine], [IMAGELIBS=-lgd])
-AC_SUBST([IMAGELIBS])
+
+AS_IF([test "x$enable_image_output" != "xno"], [
+AC_CHECK_LIB([gd], [gdImagePng], [IMAGELIBS=-lgd])
+AC_SUBST([IMAGELIBS])])
+
 AM_CONDITIONAL([HAVE_LIBGD], [test "$IMAGELIBS" = "-lgd"])
 PKG_CHECK_MODULES([CHECK], [check >= 0.9.6], [], [AC_MSG_NOTICE([testcases can not be execute without check installed])])
 AM_CONDITIONAL([HAVE_CHECK], [test "$CHECK_LIBS"])
-- 
2.1.4

